### PR TITLE
Incorporated G1 functions

### PR DESCRIPTION
**Pull request by brunoramoa**
The base of the solver was modified (cells have pointers to the faces that make up the cell).

The work of group 1 was incorporated into CRheoLAB

### CODE REVIEW

File: Face.cpp [hunk 1]
```diff
@@ -20,61 +25,225 @@ Face::Face( )
...
+    vector3 tmp_vec2 = facePoints_[1]->getPoint() - centerOfMass_;
+
+    // Computes the face normal by doing the cross product of the products (this is not the faceAreaVector)
+    vector3 areaVector_tmp = (tmp_vec1^tmp_vec2);
+    
+    // Computes the norm of the cross product between two vectors
+    double mag_vector_tmp = mag(areaVector_tmp);
+
+    // Calculates the faceAreaVector
+    areaVector_ = (areaVector_tmp/mag_vector_tmp)*area_;
```
Comment (jmnobrega) [inline Face.cpp:228]: This function should be corrected because the area vector should point from the owner to the neighbor cell

**Review by jmnobrega — CHANGES_REQUESTED** [review]
- I believe it is useful to add a.out file to .gitignore.
- It would be also useful to have 2 different branches, one for the code update and another for the G1 functions. This should to allow everyone access to the new code version, even while we are discussing the G1 implementations

File: Face.cpp (hunk 1 above)
Comment (brunoramoa) [reply Face.cpp:238]: Sugestão de resolução no (https://github.com/Computational-Rheology/CRheoLab/pull/7/commits/08c6a64288627fc6d4fae5d7e6ba86ae663e2720)

File: Point.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@ Point::Point(double x, double y, double z)
 }
 
 Point::Point()
-:   p_{0, 0 ,0}
+:   p_{-1, -1 ,-1}
```
[the remark this answers is not in the document]
Comment (brunoramoa) [reply Point.cpp:10]: the initialization to -1, is intended to let the user know that the object does not have the correct value yet. This behavior is common throughout the code

File: Mesh.cpp [hunk 3]
```diff
@@ -2,14 +2,34 @@
...
+  executablePath=executablePath.substr(0, executablePath.find_last_of("/"));
+
+  return executablePath+"/";
+}
+
+
+
 void Mesh::readMesh()
 {
     std::string pathPoints (getExecutablePath()+std::string("constant/")+std::string("polyMesh/points"));
```
[the remark this answers is not in the document]
Comment (brunoramoa) [reply Mesh.cpp:35]: Done

File: Mesh.cpp [hunk 4]
```diff
@@ -188,42 +213,104 @@ void Mesh::readNeighbour(std::string path)
...
   }
   // Close the file
   in_file.close();
 
-  nInteriorFaces_ = nNeighbours;
+  return tmp_NeighboursList;
+}
+
+
+void Mesh::readCells(std::string pathOwners, std::string pathNeighbours)
```
[the remark this answers is not in the document]
Comment (brunoramoa) [reply Mesh.cpp]: this has been updated in https://github.com/Computational-Rheology/CRheoLab/pull/7/commits/c7efaa50af87e18c15c62b2148e7a39bb5cd3816

File: Face.h [hunk 5]
```diff
@@ -5,27 +5,37 @@
 #include <vector>
 using std::vector;
 
+
+class Cell;
```
[the remark this answers is not in the document]
Comment (brunoramoa) [reply Face.h]: This is done in https://github.com/Computational-Rheology/CRheoLab/pull/7/commits/cff856746c3f436aefc2443e6fbb64310a149e6d.

**Review by jmnobrega — APPROVED** [review]

**Review by jmnobrega — APPROVED** [review]

**Review by jmnobrega — APPROVED** [review]

**Review by jmnobrega — APPROVED** [review]

**Review by jmnobrega — APPROVED** [review]

**Pull request merged**